### DISCUSSION TITLE
Problem in Binding Model

### DISCUSSION THREAD

**Discussion by jvpaul05** [wire-elements/modal#174] [Q&A]
Good Day,

Can someone help me about this problem . i have already tried public int|User $user; but it gives me error of  "syntax error, unexpected '|', expecting variable (T_VARIABLE)"

ERROR:
Typed property App\Http\Livewire\Maintenance\UserManager\EditUser::$user must be an instance of App\Models\User, int used (View: C:\Users\JAPS\Desktop\2022\Projects\ecommerce\ecommerce\resources\views\vendor\livewire-ui-modal\modal.blade.php)

BUTTON:
<x-button wire:click="$emit('openModal', 'maintenance.user-manager.edit-user', {{ json_encode(['user' => $values->id]) }})" class="text-cyan-500 dark:text-cyan-600">
      Edit
</x-button>

MODALCOMPONENT:

public User $user;

public function mount(User $user)
{$this->user = $user; }

Thank you in advance.

**Comment by PhiloNL** [comment]
Try ```public User | int $user;```
